doc: CXSPA-5921 Remove HttpClientModule from ClientAuthStoreModule - migration docs (#18457)
SAP/spartacus: docs/migration/2211_0/2211-typescript-manual.doc.md

## ActiveCartService

- `WindowRef` is now a required constructor dependency.

# Changes in the core library

## ClientAuthStoreModule

> - `HttpClientModule` has been removed from the `imports` property inside `NgModule` metadata. This should not be a problem, as long as you import `HttpClientModule` in your `app.module.ts`.

Reviewer: HttpClientModule is not anymore in our app.module.ts,
As it has been substitute by:
```
import {
  provideHttpClient,
  withFetch,
  withInterceptorsFromDi,
} from '@angular/common/http';
```
Maybe rephrasing to avoid confusion?eg: import 'httpClientModule' or inject HttpClient via 'provideHttpClient'

Reviewer 2: Good catch @FollowTheFlo !

```suggestion
- `HttpClientModule` has been removed from the `imports` property inside `NgModule` metadata. This should not be a problem, as long as you import `HttpClientModule` in your `app.module.ts`. Alternatively to _importing_ `HttpClientModule` in your `app.module.ts` you can _provide_  in `app.module.ts` the `provideHttpClient(withInterceptorsFromDi())` (optionally with `withFetch()`). For more, see https://angular.io/api/common/http/provideHttpClient
```

> - Redundant `HttpClientModule` has been removed from the `imports` property inside `NgModule` metadata. The configuration for the dependency injector of `HttpClient` should be done in one place, in the main module (usually `app.module.ts`) of the consuming application.

Reviewer 2: ```suggestion
- `HttpClientModule` has been removed from the `imports` property inside `NgModule` metadata. This should not be a problem, as long as you import `HttpClientModule` in your `app.module.ts`.
```
